packages/components/psammead-content-shift-blocker/src/index.test.jsx

```diff
@@ -0,0 +1,423 @@
+    IOInstance.fireIntersectEvent([
+      {
+        isIntersecting: false, // is not in view
+        boundingClientRect: {
+          top: -9999, // is above viewport
+          height: initialSize,
+          width: initialSize,
+        },
+      },
+    ]);
+    ROInstance.fireResizeEvent([
+      {
+        contentRect: {
+          height: initialSize + sizeIncrease, // becomes 100px larger
+          width: initialSize + sizeIncrease, // becomes 100px larger
+        },
+      },
+    ]);
+  });
+
+  expect(global.scrollTo).not.toHaveBeenCalled();
+  expect(global.pageYOffset).toEqual(initialScrollYPosition);
+});
+
+it('should adjust Y scroll position when above viewport and child content becomes larger', () => {
+  const initialScrollYPosition = 1000;
+  const expectedXPos = 0;
+  const expectedYPos = 1100;
+  const initialSize = 200; // todo change this name
+  const sizeIncrease = 100;
+  const initialPagelHeight = 5000;
+  const pageHeightDecreaseAmount = sizeIncrease;
+
+  global.pageYOffset = initialScrollYPosition;
+
+  // simulate the document becoming smaller with each call to document.body.scrollHeight
+  const mockScrollHeight = jest
+    .fn()
+    .mockReturnValueOnce(initialPagelHeight) // 1st call to document.body.scrollHeight
+    .mockReturnValueOnce(initialPagelHeight + pageHeightDecreaseAmount); // 2nd call to document.body.scrollHeight
+
+  Object.defineProperty(HTMLElement.prototype, 'scrollHeight', {
+    configurable: true,
+    get() {
+      return mockScrollHeight();
+    },
+  });
+
+  render(
+    <ContentShiftBlocker initialWidth={100} initialHeight={100}>
+      <Content />
+    </ContentShiftBlocker>,
+  );
+
+  Object.defineProperty(HTMLElement.prototype, 'scrollHeight', {
+    get() {
+      this._scrollHeight += 100; // simulate the viewport becoming larger
+      return this._scrollHeight;
+    },
+  });
+
+  act(() => {
+    // simulate above viewport and resize events
+    IOInstance.fireIntersectEvent([
+      {
+        isIntersecting: false,
+        boundingClientRect: {
+          top: -9999, // is above viewport
+          height: initialSize,
+          width: initialSize,
+        },
+      },
+    ]);
+    ROInstance.fireResizeEvent([
+      {
+        contentRect: {
+          height: initialSize + sizeIncrease, // becomes 100px larger
+          width: initialSize + sizeIncrease, // becomes 100px larger
+        },
+      },
+    ]);
+  });
+
+  expect(global.scrollTo).toHaveBeenCalledWith(expectedXPos, expectedYPos);
+  expect(global.pageYOffset).toEqual(1100);
+});
+
+xit('should load ResizeObserver polyfill if not natively suported', async () => {
+  delete global.ResizeObserver;
+
+  expect(global.ResizeObserver).not.toBeTruthy();
+
+  await render(
+    <ContentShiftBlocker>
+      <Content />
+    </ContentShiftBlocker>,
+  );
+
+  expect(true).toEqual(true); // TODO
+});
+
+xit('should load IntersectionObserver polyfill if not natively supported', async () => {
```

Comment:
Any ideas on how to test this. Mocking and checking `observe` is called does not work, I think because this module is dynamically loaded.